ext/RMagick/rmmain.c

```diff
@@ -1130,6 +1130,16 @@ Init_RMagick2(void)
         ENUMERATOR(HCLpColorspace)
         ENUMERATOR(YDbDrColorspace)
         ENUMERATORV(XyYColorspace, xyYColorspace)
+#endif
+#if defined(IMAGEMAGICK_7)
+#if defined(IMAGEMAGICK_GREATER_THAN_EQUAL_7_0_8)
+        ENUMERATOR(LinearGRAYColorspace)
+#endif
+#if defined(IMAGEMAGICK_GREATER_THAN_EQUAL_7_0_10)
+        ENUMERATOR(JzazbzColorspace)
+#endif
+#elif defined(IMAGEMAGICK_GREATER_THAN_EQUAL_6_9_10)
+        ENUMERATOR(LinearGRAYColorspace)
```

Comment:
Why do we need both this and the `7_0_8` check above? Is it because `7_0_7` *doesn't* have it? I'm not sure how heavily used the earlier point versions are, but I'm kind of wondering if we can ignore them:

```c
#if defined(DIMAGEMAGICK_GREATER_THAN_EQUAL_6_9_0)
        ENUMERATOR(LinearGRAYColorspace)
#endif
#if defined(IMAGEMAGICK_7)
        ENUMERATOR(JzazbzColorspace)
#endif
```

If people complain our solution is generally to tell people to upgrade.

Reply:
I tried to be a as precise as possible but not objecting to changing it to your suggested patch. But this will cause a compilation error when their version IM7 is below 7.0.10 or their IM6 version is below 6.9.10.

Maybe we should remove the specific IM6 and IM7 version checks and only have a define for IM7 and always require the most recent version for either IM6 or IM7? I would not object to that but I am also not someone who uses this software.

Reply:
Hmm, yeah, I guess it might be best to be the most precise for now until we have a clearer policy about which IM versions we support. I don't understand C preprocessors all that well, but could we write something like:

```c
#if defined(IMAGEMAGICK_GREATER_THAN_EQUAL_6_9_10) && !defined(IMAGEMAGICK_7)
        ENUMERATOR(LinearGRAYColorspace)
#endif
#if defined(IMAGEMAGICK_GREATER_THAN_EQUAL_7_0_8)
        ENUMERATOR(LinearGRAYColorspace)
#endif
#if defined(IMAGEMAGICK_GREATER_THAN_EQUAL_7_0_10)
        ENUMERATOR(JzazbzColorspace)
#endif
```

Reply:
I can patch this but having a separate block for the `IMAGEMAGICK_7` check looks cleaner to me.

Reply:
Okay, it's not a big deal. I was confused why there were nested `#if` blocks so thought this was a bit more intention revealing. But we can revisit it another time.